enterprise/ha/src/main/java/org/neo4j/kernel/ha/lock/forseti/ForsetiClient.java

```diff
@@ -656,6 +665,17 @@ public int id()
         return myId;
     }
 
+    private void applyWaitStrategy( Locks.ResourceType resourceType, int tries )
```

Comment:
This method among others invoked inside of `tryUpgradeToExclusiveWithShareLockHeld`. In case of transaction termination we will end up in generic `catch(Throwable e)` block that will clear wait list and 'throw RuntimeException' as result. Is it appropriate behaviour?